docs/templates/index.md

```diff
@@ -145,12 +145,20 @@ by all child processes of the agent, including SSH sessions. See the
 [Coder Terraform Provider documentation](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/agent)
 for the full list of supported arguments for the `coder_agent`.
 
-#### startup_script
+#### `startup_script`
 
 Use the Coder agent's `startup_script` to run additional commands like
 installing IDEs, [cloning dotfiles](../dotfiles.md#templates), and cloning
 project repos.
 
+**Note:** By default, the startup script is executed in the background allowing users to access the workspace before the script completes. If you want to change this, see [`startup_script_behavior`](#startup_script_behavior) below.
+
+Here are a few guidelines for writing a good startup script (more on these below):
+
+1. Use `set -e` to exit the script if any command fails and `|| true` for commands that are allowed to fail
```

Comment:
Do you have a good link for it? Maybe we can link it as an apropos. Otherwise I think it's better to K.I.S.S. We don't want to end up writing a how-to-bash. 😅

Reply:
As it's "unofficial", it's hard to find a good link. I like jevans' illustrated reasoning: https://wizardzines.com/comics/bash-errors/

But I agree, we don't want this to devolve into a full Bash scripting tutorial. At most, we link to https://tldp.org/LDP/abs/html/

Reply:
Haha, I really want to link that comic now 😆